Library/Homebrew/extend/fileutils.rb

```diff
@@ -10,6 +11,19 @@ def mktemp(prefix = name)
     prev = pwd
     tmp  = Dir.mktmpdir(prefix, HOMEBREW_TEMP)
 
+    # Make sure files inside the temporary directory have the same group as the
+    # brew instance.
+    #
+    # Reference from `man 2 open`
+    # > When a new file is created, it is given the group of the directory which
+    # contains it.
+    if (File.grpowned? HOMEBREW_BREW_FILE)
+      chown(nil, File.stat(HOMEBREW_BREW_FILE).gid, tmp)
+    else
+       #chown(nil, Etc.getpwnam(Etc.getlogin).gid, tmp)
+       chown(nil, Process.gid, tmp)
```

Comment:
@mikemcquaid @xu-cheng Either of these fallbacks allows the PR to pass the testing on your servers, so I assume that means the testbot user is not a member of the group of the brew instance it's using. However, I'm not convinced any such fallback belongs in master. An alternative would be to do nothing and allow the group to stay "wheel" rather than changing it to "staff" because "we can."

Whether this should be an error, warning, silent no-op (catch EPERM and move on), or fallback to setting "staff" is not obvious to me since it's a bit odd for the brew instance to be owned by the admin group and then allow non-admin users to run that brew instance and try to install into a Cellar, which presumably should also be owned by admin and not writable by them anyway. If actually erroring is the correct behavior, then the test servers need to have their brew instance chgrp'd to the group of the testbot user. The tests do all pass in a non-admin installation at ~standarduser/Developer/homebrew when run by standarduser.

For what it's worth, my sense is this should be an uncaught EPERM (or caught and re-thrown with a prettier error, whatever), and we shouldn't randomly be allowing non-admins to install stuff with an admin brew instance at all. If non-admins are actually supposed to be allowed to install stuff, then the brew instance should be owned by staff (or some other group they actually belong to), unless the idea is to support some kind of "advanced" (read: confused) configuration where bin/brew and Library are all admin and Cellar is all staff or something weird like that.